src/libshadertrap/src/parser.cc

```diff
@@ -254,18 +256,72 @@ bool Parser::ParseCommandAssertEqual() {
               }
               argument_identifier_2 = std::move(token);
               return true;
+            }},
+            
+            {Token::Type::kKeywordFormat, [this, &format_entries]() -> bool {
+              while (true) {
+                CommandAssertEqual::FormatEntry::Kind kind;
+                switch (tokenizer_->PeekNextToken()->GetType()) {
+                  case Token::Type::kKeywordSkipBytes:
+                    kind = CommandAssertEqual::FormatEntry::Kind::kSkip;
+                    break;
+                  case Token::Type::kKeywordTypeByte:
+                    kind = CommandAssertEqual::FormatEntry::Kind::kByte;
+                    break;
+                  case Token::Type::kKeywordTypeFloat:
+                    kind = CommandAssertEqual::FormatEntry::Kind::kFloat;
+                    break;
+                  case Token::Type::kKeywordTypeInt:
+                    kind = CommandAssertEqual::FormatEntry::Kind::kInt;
+                    break;
+                  case Token::Type::kKeywordTypeUint: {
+                    kind = CommandAssertEqual::FormatEntry::Kind::kUint;
+                    break;
+                    default:
+                      return true;
+                }
+              }
+              auto format_start_token = tokenizer_->NextToken();
+              size_t count;
+              auto maybe_count = ParseUint32("count");
+              if (!maybe_count.first) {
+                return false;
+              }
+              count = maybe_count.second;
+              format_entries.push_back(
+                  {std::move(format_start_token), kind, count});
+          
+              }
             }}},
           // BUFFERS and RENDERBUFFERS are mutually exclusive parameters
           {{Token::Type::kKeywordBuffers,
-            Token::Type::kKeywordRenderbuffers}})) {
+            Token::Type::kKeywordRenderbuffers}},
+          
+          {{Token::Type::kKeywordFormat}})) {
     return false;
   }
+  if(arguments_are_renderbuffers){
+    if(format_entries.size() != 0){
+      message_consumer_->Message(
+                    MessageConsumer::Severity::kError,NULL,
+                    "FORMAT specifier cannot be set"
+                    "for renderbuffers arguments");
+      return false;
+    }
+    parsed_commands_.push_back(MakeUnique<CommandAssertEqual>(
+    std::move(start_token),
+    std::move(argument_identifier_1), std::move(argument_identifier_2)));
+    return true;
+  }
```

Comment:
I suggest you do something like this here:

```
if (format_entries.empty()) {
   // Add a placeholder format entry specifying that the buffers should be compared byte-by-byte
   format_entires.push_back(...);
}
```

You will need to pass a token unique pointer to the constructor of the format entry. I suggest you do:

```
MakeUnique<Token>(start_token.GetType(), start_token.GetText(), ...)
```

for that.

Reply:
As agreed, we're going to do this in the executor.